R/kwic.R

```diff
@@ -88,40 +88,47 @@ kwic.corpus <- function(x, pattern, window = 5,
 #'          "Is it a bird or a plane or is it a train?")
 #' kwic(txt, c("is", "a"), valuetype = "fixed")
 #' kwic(txt, phrase(c("is", "a", "is it")), valuetype = "fixed")
-#' 
+#'
 #' toks <- tokens(txt)
 #' kwic(toks, c("is", "a"), valuetype = "fixed")
 #' kwic(toks, phrase(c("is", "a", "is it")), valuetype = "fixed")
-#' 
+#'
 #' corp <- corpus(txt)
 #' kwic(corp, c("is", "a"), valuetype = "fixed", separator = "", remove_separators = FALSE)
-#' @export 
-kwic.tokens <- function(x, pattern, window = 5, 
-                        valuetype = c("glob", "regex", "fixed"), 
-                        separator = " ", 
+#' @export
+kwic.tokens <- function(x, pattern, window = 5,
+                        valuetype = c("glob", "regex", "fixed"),
+                        separator = " ",
                         case_insensitive = TRUE, ...) {
-    
+
     valuetype <- match.arg(valuetype)
     types <- types(x)
+
+    # coerce a character vector into a dictionary
```

Comment:
This is the alternative to [my proposal](https://github.com/quanteda/quanteda/issues/1521#issuecomment-446475936) in #1521: We make each character element a dictionary key for itself, and then ensure that kwic handles dictionaries correctly. (It already does - but see below on the keywords.)

Reply:
I ended up coercing characters into a list, not a dictionary, because this changed the behaviour of sequences that were not wrapped in `phrase()`. For dictionary objects, we always consider value sequences as phrases, but not for other pattern types.